src/encoded/static/components/search.js

```diff
@@ -735,6 +735,289 @@ TypeTerm.propTypes = {
     total: PropTypes.number.isRequired,
 };
 
+class DateSelectorFacet extends React.Component {
+    constructor() {
+        super();
+
+        // Set initial React component state.
+        this.state = {
+            possibleYears: [], // all years with results
+            possibleMonths: ['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'], // all months are allowed
+            startYears: [], // possible years for the start year drop-down
+            endYears: [], // possible years for the end year drop-down
+            startMonths: [], // possible months for the start month drop-down
+            endMonths: [], // possible months for the end month drop-down
+            startYear: undefined, // chosen start year
+            startMonth: undefined, // chosen start month
+            endYear: undefined, // chosen end year
+            endMonth: undefined, // chosen end month
+            currentYear: moment().format('YYYY'), // current year
+            currentMonth: moment().format('MM'), // current month
+            currentDay: moment().format('DD'), // current day
+            activeFacet: 'date_released', // for toggle, either 'date_released' or 'date_submitted'
```

Comment:
These currentX states never change either, so probably better to just assign these just as data members of DateSelectorFacet instead of using state. I do worry about a possible invariant violation here though. If the server rendered at 23:59:59 and the browser rendered at 00:00:01 the next day, I feel like an invariant violation might be possible. Would it be possible to render this component on the server _without_ the current date, and only retrieve those and re-render after the component has mounted? Doing anything with time in code the server sees makes me nervous now.

Reply:
I moved this so that the current day, month, and year are computed when you click on the button which should prevent anything weird from happening if you leave the tab open for days and then click on the button.